### PR TITLE
[IDP-912] Add callable Jira validation github workflow

### PR DESCRIPTION
**Pull request by heqianwang**
[IDP-912] Add callable Jira validation github workflow

[IDP-912]: https://gsoftdev.atlassian.net/browse/IDP-912?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/jira.yml [hunk 1]
```diff
@@ -6,29 +6,8 @@ on:
...
-
-            $JiraIssueKey = $Matches[0]
-            $PWord = ConvertTo-SecureString -String "${{ secrets.JIRA_API_TOKEN }}" -AsPlainText -Force
-            $Credential = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList "${{ secrets.JIRA_USERNAME }}", $PWord
-
-            Write-Host "Retrieving Jira issue with number $JiraIssueKey"
-            Get-JiraIssue -Key $JiraIssueKey -Credential $Credential -ErrorAction Stop | out-null
-            Write-Host "Successfully retrieved Jira issue"
+  call-workflow-jira:
+    uses: gsoft-inc/wl-renovate-workflow/.github/workflows/reusable/jira-workflow.yml@feature/idp-912_shared_jira_workflow
```
Comment (heqianwang) [inline .github/workflows/jira.yml:10]: to be updated to `main` once we are done with testing

File: .github/workflows/reusable-jira-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            if ("${{ inputs.branch_name }}" -like "*renovate/*" -eq $True)
+            {
+              Write-Host "Skipping, renovate branch detected"
+              return
+            }
+
+            Set-JiraConfigServer -Server "${{ secrets.JIRA_URL }}"
+            if("${{ inputs.branch_name }}" -match "[A-za-z]+[_-]+[0-9]+" -eq $False)
+            {
+              throw "Branch name ${{ inputs.branch_name }} doesn't respect the required pattern."
```
Comment (meziantou) [inline .github/workflows/reusable-jira-workflow.yml:27]: Improve error message

**Review by meziantou — APPROVED** [review]

File: .github/workflows/reusable-jira-workflow.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      run: |
+            Install-Module JiraPS -Scope CurrentUser -Force
+            if ("${{ inputs.branch_name }}" -like "*renovate/*" -eq $True)
+            {
+              Write-Host "Skipping, renovate branch detected"
+              return
+            }
+
+            Set-JiraConfigServer -Server "${{ secrets.JIRA_URL }}"
+            if("${{ inputs.branch_name }}" -match "[A-za-z]+[_-]+[0-9]+" -eq $False)
```
Comment (meziantou) [inline .github/workflows/reusable-jira-workflow.yml:25]: Should it be `^feature/[a-zA-Z]+[_-][0-9]+` or `/[a-zA-Z]+[_-][0-9]+`?

Comment (meziantou) [inline .github/workflows/reusable-jira-workflow.yml:25]:
```suggestion
            if("${{ inputs.branch_name }}" -match "[a-zA-Z]+[_-][0-9]+" -eq $False)
```

File: .github/workflows/reusable-jira-workflow.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            {
+              throw "Branch name ${{ inputs.branch_name }} doesn't respect the required pattern. Example: IDP-123"
```
Comment (meziantou) [inline .github/workflows/reusable-jira-workflow.yml:27]:
```suggestion
              throw "Branch name '${{ inputs.branch_name }}' doesn't respect the required pattern ([a-zA-Z]+[_-][0-9]+). Valid branch name example: feature/IDP-123"
```

File: .github/workflows/reusable-jira-workflow.yml (hunk 3 above)
Comment (heqianwang) [reply .github/workflows/reusable-jira-workflow.yml]: good suggestion, we'll discuss with the team as a follow up on this

**Review by meziantou — APPROVED** [review]

**Pull request merged**